signup/signupController.js

```diff
@@ -106,19 +110,41 @@ define('signupController', [], function() {
 				"siteId": $scope.sharedProperties.selectedSite
 			};
 
-			SignupService.storeAppointments(data).then((response) => {
-				if (typeof response !== 'undefined' && response && response.success){
-					document.body.scrollTop = document.documentElement.scrollTop = 0;
-					$scope.appointmentId = response.appointmentId;
-					$scope.successMessage = $sce.trustAsHtml(response.message);
+			const firstName = $scope.data.firstName || '';
+			const lastName = $scope.data.lastName || '';
+			SignupService.getExistingClientAppointments(firstName, lastName).then((response) => {
+				if (typeof response !== 'undefined' && response && response.success) {
+					// response.numberExistingAppointments === false when query is empty because none exist
+					if(response.numberExistingAppointments === false || response.numberExistingAppointments['numberExistingAppointments'] === '0') {
+						$scope.hasExistingAppointment = false;
+					} else {
+						$scope.hasExistingAppointment = true;
+					}
 
-					// Send the confirmation email
-					if ($scope.data.email != null && $scope.data.email.length > 0) {
-						$scope.emailConfirmation();
+					// Only allow the client to sign up for an appointment if they don't currently have any uncancelled appointments
+					if(!$scope.hasExistingAppointment) {
+						SignupService.storeAppointments(data).then((response) => {
+							if (typeof response !== 'undefined' && response && response.success){
+								document.body.scrollTop = document.documentElement.scrollTop = 0;
+								$scope.appointmentId = response.appointmentId;
+								$scope.successMessage = $sce.trustAsHtml(response.message);
+		
+								// Send the confirmation email
+								if ($scope.data.email != null && $scope.data.email.length > 0) {
+									$scope.emailConfirmation();
+								}
+							} else {
+								NotificationUtilities.giveNotice('Failure', 'There was an error on the server while storing your appointment. Please refresh the page in a few minutes and try again.', false);
+							}
+						});
+					} else {
+						// TODO finExistingAppoitnment has already thrown an error, do we need that one?
```

Comment:
Idk what this comment means

Reply:
Sometimes I get a bit turned around on when to write a new error message vs. when to use the one returned from the service.